### ISSUE TITLE
Validate complains about fragments (false negative)

### ISSUE THREAD

**Issue by FredyC** [kamilkisiela/graphql-inspector#36]
I am not really sure what to think about this. For example I have a simple file `WareUnitFragment.graphql.ts` with this content.

```ts
import gql from 'graphql-tag'

export const WareUnitFragment = gql`
  fragment GWareUnit on WareUnit {
    id
    enum
    code
  }
`
```

Running the validate for that file outputs...

```
error in src/segments/menu/graph/WareUnitFragment.graphql.ts:

 - There can be only one fragment named GWareUnit.
 - Fragment GWareUnit is never used.
 - Fragment GAddress is never used.
 - Fragment GCustomer is never used.
 - Fragment GEmail is never used.
 ...
```

It lists all fragments used in the app. And this goes on for every file including fragments

**Comment by martyganz** [comment]
Hi @FredyC! I was able to reproduce the 'There can only be one fragment' message, but if the fragments are actually used in a document I'm not getting the other message. 

Could you provide a repro?

**Comment by FredyC** [comment]
Perhaps try to create two documents with different fragments and validate with glob against both of them? Not sure what else could I repro about it.

**Comment by martyganz** [comment]
Yeah I was thinking the same. I’ll try to reproduce it tonight or tomorrow. Thanks!

**Comment by kamilkisiela** [comment]
@FredyC I published @martyganz changes in `v0.7.2`

**Comment by FredyC** [comment]
@kamilkisiela Unfortunately, there is no change in the output. Might be useful to add a unit test for this use case. I cannot help with that at this moment, perhaps later in the week, but no promises.

**Comment by martyganz** [comment]
@FredyC just had a look but couldn't find the issue yet. I think it is because the tool looks for multiple documents and if you are not using your fragments in the same document it will give the same error for every document thus resulting in a long list for you. 

Bugfix for my stupid mistake is up.

cc @kamilkisiela

**Comment by kamilkisiela** [comment]
#52 should fix it. The issue was I assumed that including all fragments in each document would be enough to validate a document with nested fragments. That was a wrong assumption because GraphQL throws an error where those are unused.

Right now, we create a graph of dependencies between fragments and when one is used we resolve it and also its children, so then we can validate a document. This is not the best approach especially when fragments are invalid, but I guess it's good enough for now. This library is still in open development so we're all here are super super early adopters :)

**Comment by kamilkisiela** [comment]
`v0.7.3` is out

**Comment by kamilkisiela** [comment]
@FredyC let me know if that works for you.

**Comment by FredyC** [comment]
Ok, so there is definitely something different now. I still have a feeling it's a fragment related, but it only shows the following. There are no details about actual errors. There is much more of those lines there, pretty much all files with queries I think. Files with fragments only are not listed here anymore, so that's a good progress :)

The last file `Zones.graphql.ts` is perhaps a bit different as it contains queries and it exports a reusable fragment which is not actually used in that file but in others. I tried removing that fragment from that file and it failed with error `error Node does not exist: GZone`. Not very helpful as it doesn't show in which file it is expected, but that's not a huge deal.

Btw, in general, I don't think that unused fragments should be even considered as an error, perhaps a warning.

```
$ graphql-inspector validate src/**/*.graphql.ts schema/schema.json

Detected 1 invalid document:

error in src/models/AuthModel.graphql.ts:

error in src/models/UserModel.graphql.ts:

error in src/molecules/AddressSuggest.graphql.ts:

error in src/segments/zones/Zones.graphql.ts:

  - Fragment GZone is never used.
 - Fragment GZonePriceRange is never used.
 - Fragment GPrice is never used.

error Command failed with exit code 1.
```

**Comment by kamilkisiela** [comment]
@FredyC I released `v0.9.0`. I don't have a reproduction of your issue but probably enough information to guess what might solve it, I tried to get rid of all fragments and use them when an operation is being validated. You can check tests and the code here https://github.com/kamilkisiela/graphql-inspector/pull/58/.

**Comment by FredyC** [comment]
@kamilkisiela Awesome, now it seems to be truly working. Even correctly detected deprecated field use. Thank you!

**Comment by FerasAlazzeh** [comment]
I am seeing the same issue still how did you get around this @danielkcz ?
